### PR TITLE
[Homework #9] - JS-Advanced-2 by Ivan Ishchenko

### PR DESCRIPTION
**Pull request by ViVse**

### CODE REVIEW

File: homeworks/ivan-ishchenko/09-js-advanced-2/index.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
+class Serializable {
+  serialize() {
+    const serializedData = {
+      _type: this.constructor.name,
+      _data: this.#serializeData(this),
+    };
+    return JSON.stringify(serializedData, (key, value) => {
```
Comment (VitalikMelnyk) [inline homeworks/ivan-ishchenko/09-js-advanced-2/index.js:7]: "key" argument is unused. You can mark it as '_'.

File: homeworks/ivan-ishchenko/09-js-advanced-2/index.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      if (typeof value === 'number' && isNaN(value)) return 'NaN';
+      if (value === Infinity) return 'Infinity';
+      if (value === -Infinity) return '-Infinity';
+      return value;
+    });
+  }
+
+  wakeFrom(serialized) {
+    const parsedData = JSON.parse(serialized, (key, value) => {
```
Comment (VitalikMelnyk) [inline homeworks/ivan-ishchenko/09-js-advanced-2/index.js:16]: "key" argument is unused. You can mark it as '_'.

**Review by VitalikMelnyk — COMMENTED** [review]
Hello @ViVse!
I have checked your PR, and left some comments.

**Comment by ViVse** [comment]
@VitalikMelnyk, fixed all issues

**Review by VitalikMelnyk — APPROVED** [review]
Well Done @ViVse!
Approved.

**Pull request merged**